util.go

```diff
@@ -2,62 +2,89 @@ package relay
 
 import (
 	"encoding/binary"
-	"fmt"
+	"errors"
 	"io"
 
+	pb "github.com/libp2p/go-libp2p-circuit/pb"
+
+	ggio "github.com/gogo/protobuf/io"
+	proto "github.com/gogo/protobuf/proto"
+	peer "github.com/libp2p/go-libp2p-peer"
+	pstore "github.com/libp2p/go-libp2p-peerstore"
 	ma "github.com/multiformats/go-multiaddr"
+	mh "github.com/multiformats/go-multihash"
 )
 
-type singleByteReader struct {
-	r io.Reader
-}
+func peerToPeerInfo(p *pb.CircuitRelay_Peer) (empty pstore.PeerInfo, err error) {
```

Comment:
I will modify to avoid using the named arguments, @whyrusleeping doesn't like them either I think.